federation/normalize.go

```diff
@@ -146,6 +146,7 @@ func mergeSameAlias(selections []*graphql.Selection) ([]*graphql.Selection, erro
 
 	newSelections := selections[:0]
```

Comment:
For this or a future PR: This looks fishy, because of how Go slices are implemented. I think that this doesn't cause an issue with the current code, but that it has a rarely-expected effect that might surprise a future dev.

I believe that the effect of this variable initialization is that that any appends to `newSelections` will overwrite the existing values in `selections`. That might actually be fine in this code, if both:
1) The caller is okay with their `selections` variable being modified
2) Once item number i is appended newSelections, then no item at index j in selection is read, where j <= i. 
I haven't checked property 1, but property 2 looks like it holds.

Two guesses on the intention of this code:
A) It is accidental. If so, how about we change it to be `newSelections := make([]*graphql.Selection, 0, len(selections)`
B) It is intentional. This approach avoids allocating a new slice. Maybe that's an important performance property? If we think this is the case, would you add a comment for future readers, so that they don't miss the subtlety?

Example of the slice overwrite behavior: https://play.golang.org/p/cgG8JSRsaRX